azure-toolkit-libs/azure-toolkit-mysql-lib/src/main/java/com/microsoft/azure/toolkit/lib/mysql/service/MySqlFirewallRule.java

```diff
@@ -5,5 +5,41 @@
 
 package com.microsoft.azure.toolkit.lib.mysql.service;
 
+import com.azure.resourcemanager.mysql.MySqlManager;
+import com.azure.resourcemanager.mysql.models.FirewallRule;
+import com.microsoft.azure.toolkit.lib.common.task.ICommittable;
+import com.microsoft.azure.toolkit.lib.mysql.model.MySqlEntity;
+import com.microsoft.azure.toolkit.lib.mysql.model.MySqlFirewallRuleEntity;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+@AllArgsConstructor
 public class MySqlFirewallRule {
+    private final MySqlManager manager;
+    private final MySqlEntity mySqlEntity;
+    @Getter
+    private MySqlFirewallRuleEntity entity;
+
+    static MySqlFirewallRuleEntity fromFirewallRule(FirewallRule firewallRule) {
+        return MySqlFirewallRuleEntity.builder()
+            .id(firewallRule.id())
+            .name(firewallRule.name())
+            .startIpAddress(firewallRule.startIpAddress())
+            .endIpAddress(firewallRule.endIpAddress()).build();
+    }
+
+    public void delete() {
+        manager.firewallRules().delete(mySqlEntity.getResourceGroup(), this.mySqlEntity.getName(), this.entity.getName());
+    }
+
+    public ICommittable<MySqlFirewallRule> update(String startIpAddress, String endIpAddress) {
+        return () -> {
+            FirewallRule rule = manager.firewallRules().get(mySqlEntity.getResourceGroup(),
+                MySqlFirewallRule.this.mySqlEntity.getName(), MySqlFirewallRule.this.entity.getName()).update()
+                .withStartIpAddress(startIpAddress)
+                .withEndIpAddress(endIpAddress).apply();
+            MySqlFirewallRule.this.entity = fromFirewallRule(rule);
```

Comment:
need we add a check existence logic to update?

Reply:
this this class, user has already got an rule, so the rule shall be there, needn't to check existence